js/src/components/shipping-rate-section/minimum-order-card/getMinimumOrderHandlers.test.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * Internal dependencies
+ */
+import getMinimumOrderHandlers from './getMinimumOrderHandlers';
+
+describe( 'getMinimumOrderHandlers', () => {
+	let mockOnChange;
+	let handlers;
+
+	beforeEach( () => {
+		const value = [
+			{
+				id: '1',
+				country: 'US',
+				method: 'flat_rate',
+				currency: 'USD',
+				rate: 20,
+				options: {},
+			},
+			{
+				id: '2',
+				country: 'AU',
+				method: 'flat_rate',
+				currency: 'USD',
+				rate: 20,
+				options: {
+					free_shipping_threshold: 50,
+				},
+			},
+			{
+				id: '3',
+				country: 'CN',
+				method: 'flat_rate',
+				currency: 'USD',
+				rate: 25,
+				options: {
+					free_shipping_threshold: 50,
+				},
+			},
+		];
+
+		mockOnChange = jest.fn();
+		handlers = getMinimumOrderHandlers( {
+			value,
+			onChange: mockOnChange,
+		} );
+	} );
+
+	describe( 'handleAddSubmit', () => {
+		it( 'returns value with the newly added group', () => {
+			const newGroup = {
+				countries: [ 'US' ],
+				currency: 'USD',
+				threshold: 30,
+			};
+
+			const { handleAddSubmit } = handlers;
+			handleAddSubmit( newGroup );
+
+			expect( mockOnChange.mock.calls.length ).toBe( 1 );
+			expect( mockOnChange.mock.calls[ 0 ][ 0 ] ).toStrictEqual( [
+				{
+					id: '1',
+					country: 'US',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 20,
+					options: {
+						free_shipping_threshold: 30,
+					},
+				},
+				{
+					id: '2',
+					country: 'AU',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 20,
+					options: {
+						free_shipping_threshold: 50,
+					},
+				},
+				{
+					id: '3',
+					country: 'CN',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 25,
+					options: {
+						free_shipping_threshold: 50,
+					},
+				},
+			] );
+		} );
+	} );
+
+	describe( 'getChangeHandler', () => {
+		it( 'returns value updated based on changed group threshold', () => {
```

Comment:
Same as https://github.com/woocommerce/google-listings-and-ads/pull/1424#discussion_r860277610

:construction: (here and below)
"When called with a value, returns a function, which when called with a new value, calls `onChange` callback with a changed...."

:nail_care: :grey_question:
Just writing the above gives me a smell that maybe this state keeping logic may be a tad overcomplicated

Probably will get simplified and more obvious if the handlers would be separate functions
https://github.com/woocommerce/google-listings-and-ads/pull/1410/files#r860293079